explorer/error_status.go

```diff
@@ -0,0 +1,121 @@
+package explorer
+
+import (
+	"go.mondoo.com/ranger-rpc/codes"
+	"go.mondoo.com/ranger-rpc/status"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	spb "google.golang.org/genproto/googleapis/rpc/status"
+)
+
+// cnquery codes start at 100 to avoid conflicts with GRPC and Ranger RPC codes
+type ErrorStatusCode uint32
+
+const (
+	Unknown ErrorStatusCode = iota + 100
+	// NotApplicable is returned when a query or asset is not applicable to the current scan
+	NotApplicable
+	// NoQueries is returned when no queries are found in the bundle for the asset
+	NoQueries
+)
+
+func NewErrorStatusCodeFromString(s string) ErrorStatusCode {
+	switch s {
+	case "NotApplicable":
+		return NotApplicable
+	case "NoQueries":
+		return NoQueries
+	}
+	return Unknown
+}
+
+func (e ErrorStatusCode) String() string {
+	switch e {
+	case NotApplicable:
+		return "NotApplicable"
+	case NoQueries:
+		return "NoQueries"
+	default:
+		return "Unknown"
+	}
+}
+
+// cnquery codes start at 100 to avoid conflicts with GRPC and Ranger RPC codes
```

Comment:
Copy&Paste error? I do not see the 100 below in the `const` block.

Reply:
This comment is correct, it refers to https://github.com/mondoohq/ranger-rpc/blob/main/codes/codes.go. I want to avoid conflicting error codes. Updated the comment to refer to the ranger codes.